crates/ide-diagnostics/src/handlers/remove_trailing_return.rs

```diff
@@ -0,0 +1,377 @@
+use hir::{db::ExpandDatabase, diagnostics::RemoveTrailingReturn, HirFileIdExt, InFile};
+use ide_db::{assists::Assist, source_change::SourceChange};
+use syntax::{ast, AstNode, SyntaxNodePtr};
+use text_edit::TextEdit;
+
+use crate::{fix, Diagnostic, DiagnosticCode, DiagnosticsContext};
+
+// Diagnostic: remove-trailing-return
+//
+// This diagnostic is triggered when there is a redundant `return` at the end of a function
+// or closure.
+pub(crate) fn remove_trailing_return(
+    ctx: &DiagnosticsContext<'_>,
+    d: &RemoveTrailingReturn,
+) -> Diagnostic {
+    let display_range = ctx.sema.diagnostics_display_range(InFile {
+        file_id: d.file_id,
+        value: expr_stmt(ctx, d)
+            .as_ref()
+            .map(|stmt| SyntaxNodePtr::new(stmt.syntax()))
+            .unwrap_or_else(|| d.return_expr.into()),
+    });
```

Comment:
You can use `adjusted_display_range` for this (will need a rebase after https://github.com/rust-lang/rust-analyzer/pull/16459),